src/test/java/seedu/exercise/logic/parser/ParserUtilTest.java

```diff
@@ -60,6 +70,31 @@ public void parseIndex_validInput_success() throws Exception {
         assertEquals(INDEX_ONE_BASED_FIRST, ParserUtil.parseIndex("  1  "));
     }
 
+    @Test
+    public void parseIndexes_withNull_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseIndexes(null));
+    }
+
+    @Test
+    public void parseIndexes_collectionWithInvalidIndex_throwsParseException() {
+        assertThrows(ParseException.class, () ->
+                ParserUtil.parseIndexes(Arrays.asList("-1", "0")));
+    }
+
+    @Test
+    public void parseIndexes_emptyCollection_returnsEmptyList() throws Exception {
+        assertTrue(ParserUtil.parseIndexes(Collections.emptyList()).isEmpty());
+    }
+
+    @Test
+    public void parseIndexes_collectionWithValidMuscles_returnsIndexList() throws Exception {
+        ArrayList<Index> actualIndexSet = ParserUtil.parseIndexes(Arrays.asList("1", "2"));
+        ArrayList<Index> expectedIndexSet = new ArrayList<>(Arrays.asList(Index.fromOneBased(1),
+                Index.fromOneBased(2)));
```

Comment:
Can use the `CommonTestData` constant `INDEX_ONE_BASED`

Reply:
ok edited.